engine/cleanupmanager/alerts.go

```diff
@@ -0,0 +1,63 @@
+package cleanupmanager
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/riverqueue/river"
+	"github.com/target/goalert/alert"
+	"github.com/target/goalert/alert/alertlog"
+	"github.com/target/goalert/config"
+	"github.com/target/goalert/gadb"
+)
+
+type AlertArgs struct{}
+
+func (AlertArgs) Kind() string { return "cleanup-manager-alerts" }
+
+// CleanupAlerts will automatically close and delete old alerts.
+func (db *DB) CleanupAlerts(ctx context.Context, j *river.Job[AlertArgs]) error {
+	cfg := config.FromContext(ctx)
+	if cfg.Maintenance.AlertAutoCloseDays > 0 {
+		err := db.whileWork(ctx, func(ctx context.Context, tx *sql.Tx) (done bool, err error) {
+			ids, err := gadb.New(tx).CleanupMgrFindStaleAlerts(ctx, gadb.CleanupMgrFindStaleAlertsParams{
+				AutoCloseDays: int64(cfg.Maintenance.AlertAutoCloseDays),
+				IncludeAcked:  cfg.Maintenance.AutoCloseAckedAlerts,
+			})
+			if err != nil {
+				return false, fmt.Errorf("find stale alerts: %w", err)
+			}
+
+			var idsInt []int
+			for _, id := range ids {
+				idsInt = append(idsInt, int(id))
+			}
+
+			_, err = db.alertStore.UpdateManyAlertStatus(ctx, alert.StatusClosed, idsInt, alertlog.AutoClose{AlertAutoCloseDays: cfg.Maintenance.AlertAutoCloseDays})
+			if err != nil {
+				return false, fmt.Errorf("update alerts: %w", err)
+			}
+
+			return len(ids) < 100, nil
+		})
+		if err != nil {
+			return fmt.Errorf("auto close alerts: %w", err)
+		}
+	}
+
+	if cfg.Maintenance.AlertCleanupDays > 0 {
```

Comment:
They're done sequentially because we want the cleanup/delete operation to benefit from alerts that were potentially closed by auto-close (since it will ONLY delete closed alerts). However, these queries are done in "chunks" of 100 until complete, so I laid it out so that we auto-close _all_ alerts before performing the cleanup.

That said, if there were an error, River (the job queue library) will retry up to 25 times with backoff, alternatively the cleanup operation will run again in an hour anyhow.

None of this is super critical, and one of the benefits of the job queue is we can allow jobs to take as long as they need; so we don't have need for them to happen concurrently either. If either were to fail, there would be no impact to critical operations. Especially since both of these settings are 0/disabled by default.